Pre Push Script

trowacat opened the pull request:
Script to check for any private keys with a balance in the account.

This PR includes a pre-push script that will check if there are any private keys being pushed. It will then check their balance and send a warning if there are above 0.

If an issue is found it warns the dev and asks if they wish to proceed.

This script will have to be set up differently depending on the project it is being used on. In this case it is using husky.

Fixes #23

PombeirP commented on scripts/pre-push:
```diff
@@ -0,0 +1,78 @@
+  })
+}
+// get git to return files that have been altered
+result('git diff --name-only', function (err, response) {
```
I don't see the argument `${GIT_PARAMS}` being used in this script. How will git diff know what to compare against?

trowacat commented:
@PombeirP I wasn't able to get ${GIT_PARAMS} to work in the script, it seems to not pass the params in, what do you think of this solution?

PombeirP commented:
@trowacat hmm that's odd, I think I had that working locally. I'll do some testing and get back to you.

PombeirP commented:
This is what I see when I do `git push -n` (changed `prepush` to `"prepush": "git fetch --no-tags && node scripts/pre-push --loglevel warn ${GIT_PARAMS}"`):
```
0: /home/pedro/.nvm/versions/node/v9.3.0/bin/node
1: /home/pedro/src/github.com/status-im/status-github-bot/scripts/pre-push
2: --loglevel
3: warn
4: trowacat
5: git@github.com:trowacat/status-github-bot.git
```

trowacat commented:
@PombeirP You're right that worked and I was able to access it that way. Thank you for the tip. It's true, it wouldn't make sense to assume the names.

PombeirP commented:
@trowacat Thanks for the changes, we're in the right direction. However, the git diff command is still assuming that the remote head revision is the one we want to compare against. We should pass the SHA1 of the revision that is passed to the push hook (see last line of [my comment](https://github.com/status-im/status-github-bot/pull/25#issuecomment-401842823) above).

trowacat commented:
@PombeirP ah, I apologize, I was having trouble getting the SHA1 in the script but I saw I had to update husky and use their HUSKY_GIT_STDIN variable. 

Is this the correct behavior now?

trowacat commented:
@PombeirP  Thank you, I was having trouble getting the SHA1 in the script but I saw I had to update husky and use their HUSKY_GIT_STDIN variable.

Is this the correct behavior now?

PombeirP commented:
@trowacat we might as well pass both hashes (local and remote) we get from stdin, so we don't leave anything to chance.
I noticed that `package.json` was cloned to the `scripts` folder. Is that required?

trowacat commented:
@PombeirP Sure, I've added that now.

Also, no it isn't required and shouldn't be there as it's not on my local repo. I've removed it now, thank you for pointing it out.

PombeirP commented:
Merged, thanks for the contribution @trowacat!
